### PR TITLE
Inliner: updates to random inlining

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Cache the random state on the InlineStrategy instead of on the compiler
instance so that the state is reinitialized and private to each jit request.
That way the random policy evaluations made compiling one method won't
alter the evaluations for subsequent methods. This should make it somewhat
easier to do A/B comparisons under jit stress when changing the jit.

Make it possible to enable the RandomPolicy outside of stress. This may
prove useful in various randomized inline performance studies or as
a simple stress mode on its own.

Random state seed is built from an external seed value (via JitStress or
JitInlinePolicyRandom) and an internal seed value (method hash), so that
random sequences potentially differ for each method but are deterministic
across runs and changes to the jit.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@CarolEidt This should address the issue you were hitting with unexpected inlining changes in jit stress. PTAL.

cc @dotnet/jit-contrib 

This will alter the inlines done under stress, so will retest some of these.

@dotnet-bot test Windows_NT jitstress1
@dotnet-bot test Windows_NT jitstress2

**Comment by AndyAyersMS** [comment]
@mmitche seems like not all CI jobs are showing up .. I only have 4 above. Any ideas?

File: src/jit/inlinepolicy.cpp [hunk 1]
```diff
@@ -957,37 +956,24 @@ bool EnhancedLegacyPolicy::PropagateNeverToRuntime() const
...
+RandomPolicy::RandomPolicy(Compiler* compiler, bool isPrejitRoot)
     : LegalPolicy(isPrejitRoot)
     , m_RootCompiler(compiler)
     , m_Random(nullptr)
     , m_CodeSize(0)
     , m_IsForceInline(false)
     , m_IsForceInlineKnown(false)
 {
-    // If necessary, setup and seed the random state.
-    if (compiler->inlRNG == nullptr)
```
Comment (JosephTremoulet) [inline src/jit/inlinepolicy.cpp]: Can the `inlRNG` field on `Compiler` now be removed as well?

File: src/jit/jitconfigvalues.h [hunk 2]
```diff
@@ -256,6 +256,7 @@ CONFIG_INTEGER(JitInlineLimit, W("JitInlineLimit"), -1)
 CONFIG_INTEGER(JitInlinePolicyDiscretionary, W("JitInlinePolicyDiscretionary"), 0)
 CONFIG_INTEGER(JitInlinePolicyFull, W("JitInlinePolicyFull"), 0)
 CONFIG_INTEGER(JitInlinePolicySize, W("JitInlinePolicySize"), 0)
+CONFIG_INTEGER(JitInlinePolicyRandom, W("JitInlinePolicyRandom"), 0)
```
Comment (JosephTremoulet) [inline src/jit/jitconfigvalues.h]: Nit: might want comment or name to mention that this gets used to seed the generator.

File: src/jit/inlinepolicy.cpp (hunk 1 above)
Comment (AndyAyersMS) [inline src/jit/inlinepolicy.cpp]: Yes. I'll remove it.

File: src/jit/jitconfigvalues.h (hunk 2 above)
Comment (AndyAyersMS) [inline src/jit/jitconfigvalues.h]: I'll add a comment.

**Comment by AndyAyersMS** [comment]
CI needed some TLC... retrying

@dotnet-bot retest this please
@dotnet-bot test Windows_NT jitstress1
@dotnet-bot test Windows_NT jitstress2

**Comment by AndyAyersMS** [comment]
Doesn't look like the stress legs launched, so will retry.

@dotnet-bot test Windows_NT jitstress1
@dotnet-bot test Windows_NT jitstress2

**Comment by AndyAyersMS** [comment]
[Arm32 leg](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_windows_nt_prtest/22/) failed with
```
Failure to restore a package: Unable to resolve 'runtime.win7-arm.Microsoft.NETCore.CoreDisTools (>= 1.0.1-prerelease)' for 'DNXCore,Version=v5.0'.
```
retrying.

@dotnet-bot retest Windows_NT arm Cross Release Build

**Pull request merged**